platform/website/src/components/auth/auth-dialog.svelte

```diff
@@ -48,56 +50,38 @@
 	let turnstileToken = "";
 	let loggingIn = false;
 
-	let username = newField({
-		id: "username",
-		type: "text",
-		label: "Username",
-		autoComplete: "username",
-		extra: {
-			timeout: undefined as NodeJS.Timeout | undefined,
-		},
-		reload() {
-			username.touched = username.value.length > 0;
+	$: $authDialog, resetAllFields();
 
+	let usernameValue: string;
+	let usernameStatus: FieldStatus;
+	let usernameValidationTimeout: number | NodeJS.Timeout;
+	function usernameValidate(v: string): Promise<FieldStatus> {
+		return new Promise((resolve) => {
```

Comment:
Can this be an async function?

Reply:
I want to call the `resolve` function in the `setTimeout` callback, so I explicitly want to create the Promise. Let me know if there is a better way of doing it.

Reply:
Another way would be to wrap the set timeout itself in a promise, would make things look nicer I think.

Reply:
What do you mean? I want to resolve the usernameValidate Promise in the setTimeout callback

Reply:
You wrap the settimeout in a promise, and then await it

Reply:
Sorry, I still don't really get how this would work. Could you suggest some code?

Reply:
```ts
async function xyz(): string {
    await new Promise(resolve => setTimeout(() => resolve(), 2000));
    return "hi from 2s later";
}
```

Reply:
oh yeah like this, makes sense

Reply:
But how do we cancel it then?

Reply:
nvm, it seems like it just magically works